config/initializers/safe_hosts.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require 'uri'
+
+SafeHosts = Settings.web_origin.split(',').map do |url|
+  begin
+    URI.parse(url.strip).hostname.to_s
+  rescue StandardError
+    ''
+  end
+end.uniq
```

Comment:
Yes, that's right. Actually all of the reasons for why this change is needed are noted above. in the description, there are multiple different reasons. I agree with you that this array could be frozen. As for the name, since it is declared in an initializer I am just following the same precedence we have for the Settings and SSOEncryptor constants, but to your point, REDIS_CONFIG and others are declared there as well. I can make these changes.